fili-core/src/main/java/com/yahoo/bard/webservice/util/CompletedFuture.java

```diff
@@ -0,0 +1,60 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.util;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A future response which has already been completed and will return
+ * as soon as it is called.
+ *
+ * @param <E>  The type to be returned by this future
+ */
+public class CompletedFuture<E> implements Future<E> {
+    private final E item;
+    private final Throwable throwable;
+
+    /**
+     * Creates a completed future with an item to return or
+     * if a nonnull throwable is passed it it will throw an
+     * execution exception.
+     *
+     * @param item  The item to return when called.
+     * @param throwable  A non null throwable if an exception should be thrown.
+     */
+    public CompletedFuture(E item, Throwable throwable) {
```

Comment:
I see all constructor is called everywhere with `item = null`, like `new CompletedFuture<>(null, throwable)`. This makes code less readable, because people don't know that `null` means. A better approach as I learned from Rick's suggestion is

```java
public CompletedFuture(E item, Throwable throwable) {
    this.item = item;
    this.throwable = throwable;
}

public CompletedFuture(Throwable throwable) {
    this(null, throwable);
}
```

That way, you do `new CompletedFuture<>(throwable)`.